kafka-ui-react-app/src/components/common/table/TableHeaderCell/TableHeaderCell.tsx

```diff
@@ -13,38 +12,49 @@ export interface TableHeaderCellProps {
 }
 
 const TableHeaderCell: React.FC<TableHeaderCellProps> = (props) => {
-  const { title, previewText, onPreview, orderBy, orderValue, handleOrderBy } =
-    props;
+  const {
+    title,
+    previewText,
+    onPreview,
+    orderBy,
+    orderValue,
+    handleOrderBy,
+    ...restProps
+  } = props;
+
+  const isSortable = !!(orderValue && handleOrderBy);
+  const isCurrentSort = !!orderValue && orderBy === orderValue;
 
   return (
-    <S.TableHeaderCell
-      className={cx(orderBy && orderBy === orderValue && 'has-text-link-dark')}
-      {...props}
-    >
-      <span className="title">{title}</span>
+    <S.TableHeaderCell {...restProps}>
+      <S.Title
+        isSortable={isSortable}
+        isCurrentSort={isCurrentSort}
+        {...(orderValue &&
+          handleOrderBy && {
+            onClick: () => handleOrderBy(orderValue),
+            onKeyDown: (e) => e.code === 'Space' && handleOrderBy(orderValue),
+            role: 'button',
+            tabIndex: 0,
+          })}
+      >
```

Comment:
```jsx
      <S.Title
        isSortable={isSortable}
        isCurrentSort={isCurrentSort}
        onClick: handleOnClick
        onKeyDown: handleOnKeyDown
        role: 'button'
        tabIndex: 0
      >
```

I would suggest to move all these conditions inside of handleOnClick & handleOnKeyDown functions

Reply:
Rewrote this part. Still looks not ideal but better imo